src/Microsoft.ServiceFabric.Diagnostics/Metrics/Implementation/NativeTelemetry.cs

```diff
@@ -0,0 +1,34 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
+// ------------------------------------------------------------
+
+using System.Runtime.InteropServices;
+using HRESULT = System.Int32;
+#if NET
+using System.Runtime.InteropServices.Marshalling;
+#else
+using LibraryImportAttribute = System.Runtime.InteropServices.DllImportAttribute;
+#endif
+
+namespace Microsoft.ServiceFabric.Diagnostics.Metrics.Implementation
+{
+    static partial class NativeTelemetry
+    {
+        internal static IFabricMeterProvider FabricCreateMeterProvider()
+        {
+            Marshal.ThrowExceptionForHR(FabricCreateMeterProvider(out IFabricMeterProvider meterProvider));
+            return meterProvider;
+        }
+
+        [LibraryImport("FabricTelemetry")]
+        internal static
+#if NET
+        partial
+#else
+        extern
+#endif
+            HRESULT FabricCreateMeterProvider([MarshalAs(UnmanagedType.Interface)] out IFabricMeterProvider meterProvider);
```

Comment:
Could you make the indentation consistent?

Let's remove the `[MarshalAs(UnmanagedType.Interface)]`. They are redundant for interfaces. The only time when `UnmanagedType.Interface` needs to be specified explicitly is when marshalling delegates as com objects instead of function pointers.